### PR TITLE
Update nightly to conditionally build with optimizations.

### PR DESCRIPTION
**Pull request by thomasvandoren**
Update nightly to check host compiler and build without optimizations for cray
compiler. This is an experiment to see if it stabilizes the tests for this
compiler.

Also, update nightly to find make program by looking at $CHPL_NIGHTLY_MAKE
in environment, then $MAKE in environment, and finally calling chpl_make.py
script. Previously, it was defaulting to "make", which will not work on some
platforms.

### CODE REVIEW

**Comment by thomasvandoren** [comment]
I verified this by running `./util/cron/nightly -cron -hello` on my laptop and on a cray with CHPL_HOST_COMPILER=cray-prgenv-cray.

**Comment by thomasvandoren** [comment]
@ronawho , @bradcray -- Will one of you please review?

File: util/cron/nightly [hunk 1]
```diff
@@ -461,23 +468,36 @@ if ($hostplatform =~ "cygwin") {
     mysystem("cd $chplhomedir && find . -name FILES -exec rm {} \\;");
 }
 
+$hostcompiler = `$utildir/chplenv/chpl_compiler.py --host`; chomp($hostcompiler);
+
+# Setup variables to pass to all make calls.
+$make_vars_no_opt = "DEBUG=0 WARNINGS=1";
+
+# Add OPTIMIZE=1 for most environments. If using the cray programming
+# environment, disable optimizations when building the compiler.
```
Comment (ronawho) [inline util/cron/nightly]: Could use a note about why we're disabling optimizations for cray

**Comment by ronawho** [comment]
Code itself looks good.

If this doesn't seem to resolve the sporadic failures, we should revert to
building optimized if only for consistency.

I'm also curious what the impact of this will be on compile time. It shouldn't
hurt makeBinary (compiling the generated code), but resolve and a couple other
passes in our compiler will take longer.

**Pull request merged**